### PR TITLE
Feature/#26 gamearea

### PR DESCRIPTION
**Pull request by Zeavee**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/entity/AoeRadiusEntity.java [hunk 1]
```diff
@@ -8,14 +8,29 @@
 public abstract class AoeRadiusEntity extends Entity {
     private double aoeRadius;
 
+    /**
+     * A constructor for the AoeRadiusEntity
+     *
+     * @param geoPoint the location of the entity
+     */
```
Comment (RaphaelBonatti) [inline app/src/main/java/ch/epfl/sdp/entity/AoeRadiusEntity.java:15]: Thanks for adding javadoc.

File: app/src/main/java/ch/epfl/sdp/geometry/Area.java [hunk 2]
```diff
@@ -1,8 +1,124 @@
...
+    /**
+     * This method tells if the given location is inside the area
+     *
+     * @param geoPoint the location we want to know if it is inside the area
+     * @return a boolean that tells if the given location is inside the area
+     */
+    public boolean isInside(GeoPoint geoPoint) {
+        Vector vec = center.toVector().subtract(geoPoint.toVector());
+        System.out.println(vec.x());
+        System.out.println(vec.y());
```
Comment (RaphaelBonatti) [inline app/src/main/java/ch/epfl/sdp/geometry/Area.java:43]: Maybe those lines are not needed anymore.

Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/geometry/Area.java:43]: whoops

File: app/src/main/java/ch/epfl/sdp/geometry/UnboundedArea.java [hunk 3]
```diff
@@ -1,11 +1,47 @@
...
+
+    @Override
+    public void shrink(double factor) {
+        //we do not need to shrink anything here
+    }
+
+    @Override
+    public void setShrinkTransition() {
+        //there is no shrink transition here
+    }
```
Comment (RaphaelBonatti) [inline app/src/main/java/ch/epfl/sdp/geometry/UnboundedArea.java:26]: You could have created an interface that implements Area and that extends it.
Then UnboundedArea would simply be an Area and not a ShrinkableArea. ;)

Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/geometry/UnboundedArea.java:26]: You're right, but I will let it as it is for now :)

**Pull request merged**